packages/explain-plan-helper/src/utils.ts

```diff
@@ -0,0 +1,20 @@
+import type { Stage } from '.';
+
+export function getStageIndexFields(stage: Stage): Record<string, string> {
+  if (
+    stage.stage === 'EXPRESS_IXSCAN' &&
+    typeof stage.keyPattern === 'string'
+  ) {
+    // TODO (SERVER-92981): For EXPRESS stages, the keyPattern is a string, which is not valid json.
+    try {
+      return JSON.parse(stage.keyPattern);
+    } catch (e) {
+      try {
+        return JSON.parse(stage.keyPattern.replace(/(\w+):/g, '"$1":'));
```

Comment:
Just fyi, this is going to break on input like `{ key: "foo:bar" }`

It feels a bit like overkill, but `mongodb-query-parser` (or directly calling `ejson-shell-parser`) could be the right tool for the job here?